ui/components/app/edit-gas-popover/edit-gas-popover.component.js

```diff
@@ -163,6 +165,7 @@ export default function EditGasPopover({
 
     const updatedTxMeta = {
       ...updatedTransaction,
+      userEditedGasLimit,
```

Comment:
Instead of setting this using `setUserEditedGasLimit` and passing it to this component, we should just be able to calculate it here with a comparison to the original gas limit. Something like:
`userEditedGasLimit: gasLimit !== transaction.originalGasEstimate`

Reply:
@danjm What do you think, can we use `properGasLimit` instead of `transaction.originalGasEstimate` because `gasLimit` is a number and `transaction.originalGasEstimate` is a hex value and we have that `properGasLimit` set like this: `const properGasLimit = Number(hexToDecimal(transaction?.originalGasEstimate));` in `ui/hooks/gasFeeInput/useGasFeeInputs.js`? We can also use `Number(transaction.originalGasEstimate)` to change the hex value to a number.
I will wait for your response about this and than I will make the necessary changes.
Thank you

Reply:
@danjm I resolved this issue.
You can see changes here: 365a71e56615512f1a28d3b7430538fa94f58178
Is it okay now?